R/timeseriessummary-data.R

```diff
@@ -388,4 +389,86 @@ parseTSSGapTolerances <- function(reportData, timezone){
   }
   
   return(gapTolerances)
-}
\ No newline at end of file
+}
+
+#' Format TSS Correction Parameters
+#' @description formats the parameters so they are prettier to print
+#' @param corrections the corrections for the report
+#' @param timezone the timezone to parse data into
+#' @return corrections with formattedParameters field to print in report
+formatCorrectionParameters <- function(corrections, timezone) {
+  
+  corrections[["formattedParameters"]] <- ""
+  
+  if(!isEmptyOrBlank(corrections) && !isEmptyVar(corrections)) {
+    
+    for (i in seq(nrow(corrections))) {
```

Comment:
Wait, what if we just unnested it and then the entire `formatCorrectionParameters` function would just contain:

```r
parms <- corrections$parameters
corrections$parameters <- NULL
corrections <- cbind(corrections, parms)
corrections_formatted <- corrections %>% 
  rowwise() %>% 
  mutate(timezone = timezone) %>% 
  mutate(
    formattedParameters = switch(type,
                                 "Offset" = formatCorrectionsParamOffset(offset, tz),
                                 "Drift" = formatCorrectionsParamDrift(driftPoints, tz),
                                 "SinglePoint" = formatCorrectionsParamSinglePoint(value, tz),
                                 "USGSMultiPoint" = formatCorrectionsParamUSGSMultiPoint(startShiftPoints, endShiftPoints, usgsType, tz),
                                 "AdjustableTrim" = formatCorrectionsParamAdjustableTrim(upperThresholdPoints, tz),
                                 "FillGaps" = formatCorrectionsParamFillGaps(resamplePeriod, gapLimit, tz),
                                 "Deviation" = formatCorrectionsParamDeviation(deviationValue, deviationType, tz),
                                 type)) %>% 
  ungroup()
```